core/src/main/java/io/questdb/griffin/engine/functions/eq/EqVarcharFunctionFactory.java

```diff
@@ -100,7 +100,10 @@ public Function getArg() {
 
         @Override
         public boolean getBool(Record rec) {
-            return negated != Utf8s.equalsNc(constant, arg.getVarcharA(rec));
+            // argument order to equalsNc is important: the first argument can be either an inlined or
+            // a split varchar. That implementation should choose the data access pattern that is optimal
+            // for it, and the constant implementation can easily adapt to both patterns.
+            return negated != Utf8s.equalsNc(arg.getVarcharA(rec), constant);
```

Comment:
Does this optimization apply to bind variables (runtime constants) as well? This particular change seems to be around constants only.

Reply:
Added code that should work for runtime constants